docroot/profiles/custom/cgov_site/modules/custom/cgov_article/src/CgovArticleTwigExtensions.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Drupal\cgov_article;
+
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+/**
+ * Extend Drupal's AbstractExtension class.
+ */
+class CgovArticleTwigExtensions extends AbstractExtension {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getName() {
+    return 'cgov_article.CgovArticleTwigExtensions';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFilters() {
+    return [
+      new TwigFilter('spanish_format', [$this, 'spanishFormat']),
+    ];
+  }
+
+  /**
+   * Accented letters in the Spanish alphabet with their equivalent replacement.
+   *
+   * @param string $text
+   *   Accented String.
+   *
+   * @return string
+   *   Formatted String.
+   */
+  public function spanishFormat($text) {
```

Comment:
Let's not call this spanish format since it is more than for spanish. Is this based on `clean_id`? I would call it `nci_clean_id` then. It *should* have some bearing on `clean_id` since you removed `clean_id` from all the templates.

The better thing to do I think is call this function `transliterate`, remove any code having to do with `clean_id`, and in the templates below call `striptags | transliterate | clean_id` in the templates. That way you keep Drupal's clean_id for when they update their function.

Reply:
I have created a custom twig filter, so i have changed the name to ncid_transliterate as transliterate is generic name. Earlier i have used php intl extension but now i have found better solution for doing the same using PHPTransliteration class.